src/test/java/org/opensearch/security/transport/SecurityInterceptorTests.java

```diff
@@ -201,17 +195,65 @@ public <T extends TransportResponse> void sendRequest(
         assertEquals(threadPool.getThreadContext().getHeader(ConfigConstants.OPENDISTRO_SECURITY_USER_HEADER), null);
     }
 
+
+    /**
+     * Tests the scenario when remote node is on same OS version and has custom serialization enabled in node attrs
```

Comment:
```suggestion
     * Tests the scenario when remote node is on same OS version and has custom serialization not present in node attrs
```

Reply:
Tests updated as per the revised logic that relies upon the patch version check.